lib/publiccloud/provider.pm

```diff
@@ -371,7 +372,7 @@ sub terraform_apply {
         );
         upload_logs(TERRAFORM_DIR . "/$cloud_name/terraform.tfvars", failok => 1);
         script_retry('terraform init -no-color', timeout => $terraform_timeout, delay => 3, retry => 6);
-        assert_script_run("terraform workspace new $resource_group -no-color", $terraform_timeout);
+        assert_script_run("terraform workspace new ${resource_group}-${suffix} -no-color", $terraform_timeout);
```

Comment:
Why not `assert_script_run("terraform workspace new $resource_group-$suffix -no-color", $terraform_timeout);` ?